src/test/scala/io/iohk/ethereum/extvm/VMServerSpec.scala

```diff
@@ -0,0 +1,117 @@
+package io.iohk.ethereum.extvm
+
+import akka.util.ByteString
+import com.trueaccord.scalapb.GeneratedMessageCompanion
+import io.iohk.ethereum.domain.{Account, Address}
+import io.iohk.ethereum.extvm.msg.{CallContext, VMQuery}
+import io.iohk.ethereum.utils.BlockchainConfig
+import org.scalamock.scalatest.MockFactory
+import org.scalatest.{FlatSpec, Matchers}
+
+class VMServerSpec extends FlatSpec with Matchers with MockFactory {
+
+  import io.iohk.ethereum.Fixtures.Blocks._
+  import Implicits._
+
+  "VMServer" should "start and await hello message" in new TestSetup {
+    inSequence {
+      (messageHandler.awaitMessage(_: GeneratedMessageCompanion[msg.Hello])).expects(*).returns(helloMsg)
+      (messageHandler.awaitMessage(_: GeneratedMessageCompanion[msg.CallContext])).expects(*).throwing(new RuntimeException) // connection closed
+      (messageHandler.close _).expects()
+    }
+    vmServer.run()
+    vmServer.processingThread.join()
+  }
+
+  it should "handle incoming call context msg" in new TestSetup {
```

Comment:
`it should "handle incoming call context msg and respond with a call result" - since that's also tested here